src/js/NasaTable/NasaTable.js

```diff
@@ -48,46 +62,70 @@ class NasaTable extends Component {
     this.setState({ rowsPerPage: rows, visibleRows, page: 0 });
   };
 
-  fillCells = (cell, i) => {
-    if (i !== 0) {
-      return <TableCell key={i}>{cell}</TableCell>;
+  setDataToShow = ({ rowsPerPage, page, data, sortKey, isDescendingSort }) => {
+    const from = rowsPerPage * page;
+    const visibleRows = this.getDataToShow(data.table, from, rowsPerPage, sortKey, isDescendingSort);
+    this.setState({ data, visibleRows });
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.isDescendingSort !== this.state.isDescendingSort) {
+      const { rowsPerPage, page, data, sortKey, isDescendingSort } = this.state;
+      this.setDataToShow({ rowsPerPage, page, data, sortKey, isDescendingSort });
```

Comment:
is it OK to write like that for better readability (all the function arguments are clear) or should I use just ` this.setDataToShow(this.state)`?

Reply:
If you need all values from the state you can use spread operator (e.g. `{ ...this.state }`). This is an equivalent to Object.assign. But you should be aware of using only primitive values here - Object.assign makes only a shallow copy